plugins/modules/user_role_workflow_manager.py

```diff
@@ -963,6 +963,33 @@ def validate_string_field(self, field_value, regex, error_message, error_message
         if field_value and not regex.match(field_value):
             error_messages.append(error_message)
 
+    def validate_password(self, password, error_messages):
+        """
+        Validate password from password and append error message if it does not match the criteria.
+        """
+
+        if password:
+            password_is_valid = False
+            password_regex_msg = "Password must be 8 to 20 characters long and should contain characters from at \
+least three of the following classes: lowercase characters, uppercase characters, digits and special characters."
+
+            self.log(password_regex_msg, "DEBUG")
+            password_regexs = [
+                re.compile(r'^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?!.*[\W_]).{8,20}$'),
+                re.compile(r'^(?=.*[a-z])(?=.*[A-Z])(?=.*[\W_])(?!.*\d).{8,20}$'),
+                re.compile(r'^(?=.*[a-z])(?=.*\d)(?=.*[\W_])(?!.*[A-Z]).{8,20}$'),
+                re.compile(r'^(?=.*[A-Z])(?=.*\d)(?=.*[\W_])(?!.*[a-z]).{8,20}$'),
+                re.compile(r'^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[\W_]).{8,20}$')
+            ]
+
+            for password_regex in password_regexs:
+                if password_regex.match(password):
+                    password_is_valid = True
+                    break
+
+            if not password_is_valid:
+                error_messages.append(password_regex_msg)
```

Comment:
Can we print a debug message?
`    self.log("Password validation failed: {0}".format(password_criteria_message), "DEBUG")
`

Reply:
Updated the code accordingly.